lib/api/core/models/repositories/tokenRepository.js

```diff
@@ -105,32 +121,39 @@ class TokenRepository extends Repository {
       encodedToken = jwt.sign(
         { _id: user._id },
         this.kuzzle.config.security.jwt.secret,
-        options);
+        signOptions);
     }
     catch (err) {
       return securityError.rejectFrom(err, 'generation_failed', err.message);
     }
 
-    const token = new Token({
-      _id: user._id + '#' + encodedToken,
-      jwt: encodedToken,
-      userId: user._id,
-      ttl: expiresIn,
-      expiresAt: Date.now() + expiresIn
-    });
-
-    return this.persistToCache(token, {ttl: expiresIn / 1000})
-      .then(() => {
-        const existingToken = this.kuzzle.tokenManager.getConnectedUserToken(
-          user._id,
-          request.context.connection.id);
+    // @todo test unit
+    return this.persistForUser(encodedToken, user._id, parsedExpiresIn);
+  }
 
-        if (existingToken) {
-          this.kuzzle.tokenManager.refresh(existingToken, token);
-        }
+  /**
+   * Persists a token in Redis cache
+   *
+   * @param {String} encodedToken - Encoded token
+   * @param {String} userId - User ID
+   * @param {Number} ttl - TTL in ms (-1 for infinite duration)
+   *
+   * @returns {Promise}
+   */
+  persistForUser (encodedToken, userId, ttl) {
+    // @todo test unit
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)